# CyberCast
## Phase 2.1
### Experiment / Training

In [1]:
import os
os.chdir('../..')
import pandas as pd
import json
import matplotlib.pyplot as plt
import seaborn as sns


### 1-5. Objective and Configuration
Phase 2.1 explored class weights, loss functions, and architectural modifications to stabilize training.



### 6-8. Experiment Comparison and Winner Selection
Actual validation configuration winner:
- Base LSTM 64
- pos_weight = 4.0
- Validation PR-AUC = 0.5087
- Validation F1 = 0.3727


In [2]:

exp_path = 'results/phase2_1/experiment_comparison.csv'
if os.path.exists(exp_path):
    df = pd.read_csv(exp_path)
    display(df)
else:
    print("Experiment comparison artifact unavailable.")


,Config,Params,Val_PR_AUC,Val_F1,Val_Precision,Val_Recall,Val_ROC_AUC,Val_Accuracy,Best_Epoch,Epochs_Trained,Val_FPR,Val_FNR
0,E1_pw=4.0,56129,0.508715,0.372662,0.723464,0.250969,0.692119,0.797020,1,11,0.030331,0.749031
1,E2_BCE_bestpw,56129,0.508715,0.372662,0.723464,0.250969,0.692119,0.797020,1,11,0.030331,0.749031
2,E3_LSTM_64_32,33313,0.507181,0.312112,0.785156,0.194767,0.695475,0.793762,1,11,0.016850,0.805233
3,E2_Focal_g1,56129,0.506879,0.351729,0.730887,0.231589,0.693478,0.794926,1,11,0.026961,0.768411
4,E1_pw=5.19,56129,0.504205,0.397386,0.610442,0.294574,0.689991,0.785382,1,11,0.059436,0.705426
5,E1_pw=3.0,56129,0.503784,0.307080,0.857778,0.187016,0.692299,0.797253,1,11,0.009804,0.812984
6,E1_pw=auto,56129,0.503662,0.397126,0.609218,0.294574,0.690621,0.785149,1,11,0.059743,0.705426
7,E2_Focal_g2,56129,0.503276,0.351391,0.718563,0.232558,0.692830,0.793762,1,11,0.028799,0.767442
8,E3_LSTM_LNorm,56257,0.502369,0.375676,0.620536,0.269380,0.693065,0.784916,1,11,0.052083,0.730620
9,E3_LSTM_Attn,22914,0.502112,0.231293,0.944444,0.131783,0.692186,0.789572,1,11,0.002451,0.868217



### 9-10. Locked Threshold and Artifacts
Locked threshold = 0.19

**RESULT STATUS:** HISTORICAL RESULT
